packages/app/lib/models/sightings.dart

```diff
@@ -82,32 +95,14 @@ class Sighting {
       this.longitude = longitude;
     }
 
-    List<DocumentId>? speciesIds;
-    if (species != null && species.isEmpty) {
-      this.species = null;
-      speciesIds = [];
-    } else if (species != null) {
-      this.species = species.first;
-      speciesIds = [species.first.id];
+    if (updatedSpecie != null) {
+      this.species = updatedSpecie;
     }
 
-    List<DocumentId>? localNameIds;
-    if (localNames != null && localNames.isEmpty) {
-      this.localName = null;
-      localNameIds = [];
-    } else if (localNames != null) {
-      this.localName = localNames.first;
-      localNameIds = [localNames.first.id];
+    if (updatedLocalName != null) {
+      this.localName = updatedLocalName;
```

Comment:
This maintains the previous behavior, but is that what we want?

right now this would be a little broken if say we removed a local name or species from a sighting, but not sure if the UI (eventually?) exposes a way to unset the species or local name

Reply:
turns out that it's possible to unset the species and/or local name in the UI, so keeping this behavior was actually a bug. Addressed via https://github.com/p2panda/meli/pull/60/commits/5e3f7dae9116f808a0f38b3451a6532600b59996

Reply:
ah wait i see what's happening. the `update()` can take a partial state - so what was originally implemented is correct 🙃

Reply:
alright proper behavior restored now: https://github.com/p2panda/meli/pull/60/commits/ffcb3b5abdf137c55510e05982ad0960e3ceef2b